public/app/plugins/datasource/opentsdb/components/OpenTsdbQueryEditor.tsx

```diff
@@ -0,0 +1,167 @@
+import { css } from '@emotion/css';
+import React, { useState } from 'react';
+
+import { GrafanaTheme2, QueryEditorProps, textUtil } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+
+import OpenTsDatasource from '../datasource';
+import { OpenTsdbOptions, OpenTsdbQuery } from '../types';
+
+import { DownSample } from './DownSample';
+import { MetricSection } from './MetricSection';
+
+export type OpenTsdbQueryEditorProps = QueryEditorProps<OpenTsDatasource, OpenTsdbQuery, OpenTsdbOptions>;
+export type OpenTsdbErrors = {
+  downsampleInterval?: string;
+  filters?: string;
+  tags?: string;
+};
+
+export function OpenTsdbQueryEditor({
+  datasource,
+  onRunQuery,
+  onChange,
+  query,
+  range,
+  queries,
+}: OpenTsdbQueryEditorProps) {
+  const styles = useStyles2(getStyles);
+
+  // errors will be updated so they need to be state
+  // They are only displayed for tags, filters and downsampling.
+  // There is evidence that metrics had errors at one point,
+  // but there is no evidence it is being handled in the fromt end,
+  // just errors delivered from the backend
+  const [errors /*,setErrors*/] = useState<OpenTsdbErrors>({});
+
+  // Xaggregators: any;
```

Comment:
Do we need those comments starting with `X`?

Reply:
Nope! that is just my list in the code of things that I have handled and I use them as a checkpoint for regressions. I will remove them in the future.